src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
too nesting, it will be more readable to do like this:

```
people.filter(p => {
 const isWithSon = onlyWithSon ? p.sex === 'm' : true;
 
 return people.some(m => m.name
      === p.mother) && isWithSon
})
```